workspaces/adoption-insights/plugins/adoption-insights/src/components/Templates/Templates.tsx

```diff
@@ -0,0 +1,164 @@
+
+import CardWrapper from '../CardWrapper';
+import { TEMPLATE_TABLE_HEADERS } from '../../utils/constants';
+import TableFooterPagination from '../CardFooter';
+import { useTemplates } from '../../hooks/useTemplates';
+
+const Templates = () => {
+  const [page, setPage] = React.useState(0);
+  const [limit] = React.useState(20);
+  const [rowsPerPage, setRowsPerPage] = React.useState(3);
+  const entityLink = useRouteRef(entityRouteRef);
+
+  const { templates, loading } = useTemplates({ limit });
+
+  const handleChangePage = React.useCallback(
+    (_event: unknown, newPage: number) => {
+      setPage(newPage);
+    },
+    [],
+  );
+
+  const handleChangeRowsPerPage = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      setRowsPerPage(+event.target.value);
+      setPage(0);
+    },
+    [],
+  );
+
+  const visibleTemplates = React.useMemo(() => {
+    return templates.data?.slice(
+      page * rowsPerPage,
+      page * rowsPerPage + rowsPerPage,
+    );
+  }, [templates, page, rowsPerPage]);
+
+  if (!visibleTemplates || visibleTemplates?.length === 0) {
+    return (
+      <CardWrapper title="Top templates">
+        <Box
+          display="flex"
+          justifyContent="center"
+          alignItems="center"
+          height={200}
+        >
+          <Typography>No results for this time range.</Typography>
+        </Box>
+      </CardWrapper>
+    );
+  }
+
+  return (
+    <CardWrapper title={`Top ${rowsPerPage} templates`}>
+      <Table aria-labelledby="Catalog entities" sx={{ width: '100%' }}>
+        <TableHead>
+          <TableRow>
+            {TEMPLATE_TABLE_HEADERS.map(header => (
+              <TableCell
+                key={header.id}
+                align="left"
+                sx={{
+                  borderBottom: theme => `1px solid ${theme.palette.grey[300]}`,
+                }}
+              >
+                {header.title}
+              </TableCell>
+            ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {loading ? (
+            <TableRow>
+              <TableCell colSpan={TEMPLATE_TABLE_HEADERS.length} align="center">
+                <CircularProgress />
+              </TableCell>
+            </TableRow>
+          ) : (
+            visibleTemplates?.map(template => (
+              <TableRow
+                key={template.entityRef}
+                sx={{
+                  '&:nth-of-type(odd)': { backgroundColor: 'inherit' },
+                  borderBottom: theme => `1px solid ${theme.palette.grey[300]}`,
+                }}
+              >
+                <TableCell sx={{ width: '50%' }}>
+                  <Link
+                    component="a"
+                    href={entityLink({
+                      kind: parseEntityRef(template.entityRef).kind,
+                      namespace: 'default',
+                      name: parseEntityRef(template.entityRef)?.name,
```

Comment:
nit: 
parseEntityRef method is used multiple places in this file and in Techdocs.tsx, can you move it to a local variable in the top of files, so you can refer that variable to get kind and name information?.